app/master/build.py

```diff
@@ -229,19 +228,16 @@ def complete_subjob(self, subjob_id, payload=None):
             self.mark_failed('Error occurred while completing subjob {}.'.format(subjob_id))
             raise
 
-    def _get_atom_artifact_sys_path(self, subjob, atom_id):
-        return os.path.join(
-            self._build_results_dir(),
-            subjob.get_atom_artifact_name(atom_id),
-        )
-
     def _parse_payload_for_atom_exit_code(self, subjob_id):
         subjob = self.subjob(subjob_id)
         for atom_id in range(len(subjob.atoms)):
-            atom_exit_code_file_sys_path = os.path.join(
-                self._get_atom_artifact_sys_path(subjob, atom_id),
-                Subjob.EXIT_CODE_FILE,
+            artifact_dir = BuildArtifact.artifact_directory(
+                self.build_id(),
+                subjob.subjob_id(),
+                atom_id,
+                result_root=Configuration['results_directory']
             )
+            atom_exit_code_file_sys_path = os.path.join(artifact_dir, BuildArtifact.EXIT_CODE_FILE)
```

Comment:
Here you're calling a static method of `BuildArtifact` and immediately join it with `BuildArtifact.EXIT_CODE_FILE`. How you think about making some helpers so few components actually need to know `BuildArtifact.EXIT_CODE_FILE`?